core/src/main/java/org/apache/iceberg/SchemaParser.java

```diff
@@ -42,6 +42,7 @@ private SchemaParser() {}
   private static final String STRUCT = "struct";
   private static final String LIST = "list";
   private static final String MAP = "map";
+  private static final String VARIANT = "variant";
```

Comment:
Types don't typically go here. Maybe add this to `fromPrimitiveString`. It fits better there, even if it isn't a primitive.

Reply:
I was thinking of that. But PrimitiveHolder calls fromPrimitiveString() which is used in PrimitiveType class. Not sure if it would cause an issue in serialization. 

> /** Replacement for primitive types in Java Serialization. */
> class PrimitiveHolder implements Serializable {
>   private String typeAsString = null;
> ...
>   Object readResolve() throws ObjectStreamException {
>     return Types.fromPrimitiveString(typeAsString);
>   }

Reply:
And also fromPrimitiveString() is returning PrimitiveType. I think it's better not to mix variant with primitive. I create typeFromTypeString(String typeString) to handle both cases. Let me know how that looks.